source/sync/data-model/update-data-model/update-data-model-production.txt

```diff
@@ -0,0 +1,431 @@
+
+- Non-breaking changes do not create a new version. They're added to your
+  current version and don't require additional handling.
+- Schema versioning changes create a new backward-compatible schema version. App
+  Services provides a server-initiated migration path for existing clients.
+- Breaking changes set a new minimum schema version. They break compatibility
+  with all previous versions and require you to manually handle reconnecting
+  clients at the new minimum version.
+
+.. important:: Schema Versioning is Incompatible with Development Mode
+
+   You must :ref:`disable Development Mode <enable-development-mode>` to use
+   schema versioning in your app. While Development Mode is enabled, App Services disables schema versioning considers schema versioning changes to be breaking changes.
+
+   If you enable Development Mode in an app with versioned schemas, the server
+   deletes all existing versions. This will break any clients using a versioned
+   schema.
+
+Non-Breaking Changes
+~~~~~~~~~~~~~~~~~~~~
+
+Once your app is in production, you can make :ref:`non-breaking changes
+<non-breaking-changes-overview>` to your server-side schema, then update the
+client models without additional action.
+
+The following are considered non-breaking changes when made to your server-side schema:
+
+- Adding a new object type
+- Adding a new property
+- Adding a default value on a property
+
+When deployed, the change is applied to the current schema version *without*
+creating a new version. Instead, the change is added to the current version and
+does not require additional handling.
+
+.. note:: Clients on Lower Schema Version Cannot Access Added Elements
+
+   Because non-breaking changes are added to the current version, clients can
+   only reference added elements (objects or properties added as part of the
+   non-breaking changeset) if they connect at the current schema version or
+   higher.
+
+If schema versioning is disabled in your app, non-breaking changes are handled
+the same. They are applied to the same single schema with no version and clients
+must update their models to access the new elements.
+
+Schema Versioning Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Schema versioning changes <schema-versioning-overview>`
+are a special subset of breaking changes. You can make schema versioning changes
+to your server-side schema, and the server creates a new backward-compatible
+version. You can then update the client models to the new schema without having
+to perform a client reset. Note that you cannot update your schema with schema
+versioning changes using the {+cli+} or automated deploy with GitHub. You must
+use the App Services UI.
+
+The following are considered schema versioning changes when made to your
+server-side schema:
+
+- Removing an existing object type
+- Removing an existing property
+- Changing whether an existing property is optional or required
```

Comment:
Do you think its worth adding some note about how this is the base set of changes we have implemented but the goal would be to support more transformations down the road? Might not be worth it, but just throwing out an idea

Reply:
I'd love to, but referring to future functionality in the docs (even in vague ways that don't guarantee anything) almost always opens up a can of worms. Probably more appropriate for something like a blog post.